### ISSUE TITLE
The accuracy is not up to the accuracy in paper and Jitter during rerank

### ISSUE THREAD

**Issue by kaiyi98** [bytedance/R2Former#16]
Hi！
I have trained the r2former, but I can't get the same result as paper.
Firstly, I run thr train_global_retrieval.sh (train_batch_size=12). I get the best R@1 over 79.

Then, I run  the first command of train_reranking.sh (train_batch_size=24, rerank_batch_size=6). I get the best R@1=87.432, and the rerank recall changes as follows
![QQ截图20240229102739](https://github.com/bytedance/R2Former/assets/89056209/8edd9821-7bf3-4a07-8c76-699d438d17ad)

Lastly,  I run  the  second command of train_reranking.sh (train_batch_size=24, rerank_batch_size=6). I get the best R@1=87.297, and the rerank recall changes as follows
![QQ截图20240229102754](https://github.com/bytedance/R2Former/assets/89056209/aea781c9-e4d3-47a9-b011-51a205a7d57a)
and the last model rerank R@1 is lower than best model, but the global retrieval is the opposite, as follows
best model
![image](https://github.com/bytedance/R2Former/assets/89056209/555ba3a7-e959-43fc-9afb-36d3f46aaea0)
last model
![image (1)](https://github.com/bytedance/R2Former/assets/89056209/3b3d8ae9-0868-42cd-a1a2-0b74b2c42490)
Q1: Why is this happening?

Other parameters remain unchanged. I train the code on 2 RTX3090.
Q2: Why The accuracy is not up to the accuracy in paper and Jitter during rerank?

Thanks!

**Comment by Jeff-Zilence** [comment]
This is normal if you are changing the overall batch size but keep the lr unchanged. You might tune the lr to get the same results, but we did not try using 2 GPUs. Unfortunately, the lr scaling rule does not apply to Adam optimizer and you might need to try different numbers to figure it out. You might also check the version of python packages and torch versions carefully.

The first stage model does not matter that much. You should be able to get a similar performance if you just load our pre-trained stage-1 model.

**Comment by nuozimiaowu** [comment]
Execuse me,during the training of R2Former，I run thr train_global_retrieval.sh, with the minime batch size, but the the thread is killed during my training, maybe because It out of the range of my CPU memory（120G），I wants to konw how much memory allows  the program run, or it is other things cause the problem.

**Comment by Jeff-Zilence** [comment]
This has nothing to do with CPU memory. The global retrieval model does not require much memory. You can check the original repo https://github.com/gmberton/deep-visual-geo-localization-benchmark and see if you can run their training script.
